### PR TITLE
allow sending in passport instance as option and export passport

### PR DESCRIPTION
**Pull request by peterwillis**
### Description
Passport is designed to be modular, it's better not too assume this module will be initializing it unless you are giving it it's own instance by doing `new passport.Passport()`. This modification means you can pass in an existing passport that is assumed to be initialized, and then also export it so it can be modified. I know that passport is a singleton, but if you call initialize twice you can end up with bugs because there could be two instances flying around.

#### Related issues

- connect to <link_to_referenced_issue>

### Checklist

- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

File: lib/oauth2-loopback.js [hunk 1]
```diff
@@ -52,11 +52,15 @@ module.exports = function(app, options) {
 
   // Default to true
   var session = (options.session !== false);
-
-  app.middleware('auth:before', passport.initialize());
-  if (session) {
-    app.middleware('auth', passport.session());
+  if (options.passport) {
+    passport = options.passport
```
Comment (dhmlau) [inline lib/oauth2-loopback.js]: missing `;`

**Comment by dhmlau** [comment]
@peterwillis , thanks for your contribution.  Your code looks fine to me, however, I'm not familiar with this component to approve your PR.  I can help you with the first few things: 
- Could you please sign the CLA: https://cla.strongloop.com/agreements/strongloop/loopback-component-oauth2
- There's a commit linter error because the commit message is over 50 characters.  Please see the [commit message guideline](http://loopback.io/doc/en/contrib/git-commit-messages.html).

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Pull request closed without merging**